src/gravel/controllers/resources/storage.py

```diff
@@ -127,3 +127,11 @@ async def _update(self) -> None:
             by_name[p.name] = pool
         self._state.pools_by_name = by_name
         self._state.pools_by_id = by_id
+
+
+_storage = Storage()
+
+
+def get_storage() -> Storage:
+    global _storage
+    return _storage
```

Comment:
Really these are still declared at a module state.. (referring to _inventory too).

I don't think we need to block this PR on this, but I think we should reconsider how we are keeping and passing state/context around.

Reply:
they are, but they are no longer at the `resources/__init__.py` level, which cause a whole lot of different problems during imports.

And, aside from having everything being initiated on `GlobalState()` and then having only GlobalState as a single global state class, I don't see how we can have these things initialized and ready to go to be used in the fastapi routes.

Unfortunately, the way these are created makes it  impossible (at least I haven't found an alternative) to pass them context to be handled on a file-by-file/class-by-class basis.

Reply:
Right, having the GlobalState handle everything is something I think that is worth considering.

Reply:
Lets attempt that in the near future then :)

Just filed https://github.com/aquarist-labs/aquarium/issues/276 to track that.